exo/inference/mlx/sharded_inference_engine.py

```diff
@@ -53,24 +69,84 @@ async def decode(self, shard: Shard, tokens) -> str:
     await self.ensure_shard(shard)
     tokens = await asyncio.get_running_loop().run_in_executor(self.executor, self.tokenizer.decode, tokens)
     return tokens
+
+  async def save_checkpoint(self, shard: Shard, path: str):
```

Comment:
Should also be an abstract method in `InferenceEngine`

Reply:
Agreed

Reply:
Did you add this?

Reply:
Bump.

Reply:
Ah, I responded to the wrong conversation. I think checkpoint loading makes sense for pure inference engine use cases but saving only makes sense if we expect the weights to change, and so could only work if training is a capability of the inference engine in question. I can add the abstractmethod but it's a design decision that will have to be worked around later if using inference engines that don't support training

Reply:
I'll put it in for now, and we can cross that bridge as we come to it I suppose